ntp-client/src/Network/NTP/Client/Query.hs

```diff
@@ -205,17 +205,10 @@ runNtpQueries ioManager tracer protocol netSettings localAddr destAddrs
         Socket.setSocketOption socket Socket.ReuseAddr 1
         Socket.bind socket (Socket.addrAddress localAddr)
         inQueue <- atomically $ newTVar []
-        withAsync timeout $ \timeoutT ->
-          withAsync (receiver socket inQueue ) $ \receiverT ->
-            -- mask async exceptions to guarantee that the other threads are
-            -- cancelled correctly.  'timeoutT' and 'receiverT' threads were
-            -- started using 'withAsync', so they will be terminated when the
-            -- callbak either returns or errors.
-            mask $ \unmask ->
-              async (unmask $ send socket) >>= \senderT -> unmask $
-                waitCatch senderT >>= \case
-                  Left e  -> throwIO e
-                  Right _ -> void $ waitAny [timeoutT, receiverT]
+        withAsync timeout $ \timeoutAsync ->
+          withAsync (receiver socket inQueue) $ \receiverAsync -> do
+            send socket
+            void $ waitAny [timeoutAsync, receiverAsync]
```

Comment:
The idea was to also guard against a send that is blocked indefinitely with the timeout.
I don't know if send can block forever.

Reply:
That would be odd. We send small number of small packets. This was protecting against async exceptions being thrown from outside, not the timeouts - timeout thread woudld need to throw an async exception but it just terminates.